skyportal/facility_apis/observation_plan.py

```diff
@@ -451,6 +451,8 @@ def generate_plan(
                 request.payload.get("ra_slice_min", 0),
                 request.payload.get("ra_slice_max", 360),
             ],
+            # use only primary grid
+            'doUsePrimary': request.payload.get("use_primary", False),
```

Comment:
I think that what we wanted is that we can:
1. First create a plan only with the primary, which means being able to turn off the secondary
2. Target only the secondary when we managed to cover the primary
3. Maybe do both.

Not sure what @tahumada wanted that now the primary is an option, but rather the opposite that the secondary is turned off by default.

Reply:
Or are you telling me that this doUsePrimary means that you ONLY use the primary??? This is confusing hahaha

Reply:
Yes it just uses the primary. It is a quick way to avoid overlapping tiles to save time. This brings us to consistency with O3, the second requires gwemopt changes.

Reply:
Just wondering if in this PR we can also add the doUseSecondary arg and make it so that on the frontend you can pass both to True (all grid), just primary, or just secondary

Reply:
@Theodlz yes added.

Reply:
Do you want me to add the frontend for that too?

Reply:
https://github.com/skyportal/skyportal/pull/4241/files#diff-8c6d218804fafa52e37ebbea006afc0200ecff0d2c71d95d76df6176462496c2R992

It is one of the options on the front-end. Feel free to make changes.

Reply:
Right my apologies I forgot about the rjsf schemas. Looks good.